src/main/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresDockingControlImpl.java

```diff
@@ -158,20 +174,71 @@ private boolean _isAccept() {
                                                  getShape());
     }
 
-    public static Point2D calculateCandidateLocation(WiresShape shape,
-                                                     WiresContainer parent) {
-        final Point2D location = shape.getControl().getParentPickerControl().getShapeLocation();
-        final Point2D trgAbsOffset = parent.getContainer().getComputedLocation();
-        return new Point2D(location.getX() - trgAbsOffset.getX(),
-                           location.getY() - trgAbsOffset.getY());
+    private Point2D calculateCandidateLocation(WiresShape shape, WiresContainer parent) {
+        return calculateCandidateLocation(shape, getCloserMagnet(shape, parent));
+    }
+
+    private Point2D calculateCandidateLocation(WiresShape shape, WiresMagnet shapeMagnet) {
+        final Point2D location = new Point2D(shapeMagnet.getX(), shapeMagnet.getY());
+        final BoundingBox box = shape.getPath().getBoundingBox();
+        final double newX = location.getX() - (box.getWidth() / 2);
+        final double newY = location.getY() - (box.getHeight() / 2);
+        return new Point2D(newX, newY);
+    }
+
+    private WiresMagnet getCloserMagnet(WiresShape shape,
+                                        WiresContainer parent) {
+        final WiresShape parentShape = (WiresShape) parent;
+        final MagnetManager.Magnets magnets = parentShape.getMagnets();
+        final double shapeWidth = shape.getPath().getBoundingBox().getMaxX();
+        final double shapeHeight = shape.getPath().getBoundingBox().getMaxY();
+
+        int magnetIndex = -1;
+        Double minDistance = null;
+        //not considering the zero magnet, that is the center.
+        for (int i = 1; i < magnets.size(); i++) {
+            WiresMagnet magnet = magnets.getMagnet(i);
+            final Point2D shapeLocation = shape.getComputedLocation();
+            final double magnetX = magnet.getControl().getLocation().getX();
+            final double magnetY = magnet.getControl().getLocation().getY();
+            double distance = abs((abs(shapeLocation.getX()) - abs(magnetX)))
```

Comment:
Not sure if really makes sense using the shape's width/height... I mean I would consider always the shape's bounding box center point as reference, which you can obtain outside the loop by calculating once, and then by just doing trigonometry calculate distance between each of the magnets point... you can use methods already present in the `Geometry` class, for example `Geometry.distance()`, no need to do the operations again here. 
Anyway just brainstorming, not sure if that makes sense or the implications for it, you know better :) Thx